app/controllers/report_controller/dashboards.rb

```diff
@@ -411,36 +411,28 @@ def db_available_widgets_xml
     end
     @available_widgets.sort_by! { |w| [w.content_type, w.title.downcase] }
 
-    xml = REXML::Document.load("")
-    xml << REXML::XMLDecl.new(1.0, "UTF-8")
-    # Create root element
-    root = xml.add_element("complete")
-    opt = root.add_element("option", {"value"=>"","img_src"=>"/images/icons/24/add_widget.png"})
-    opt.text = "Add a Widget"
-    opt.add_attribute("selected","true")
-    @available_widgets.each do |w|
-      unless col_widgets.include?(w.id) || !w.enabled
-        image, tip = case w.content_type
-                      when "rss"
-                        ["rssfeed", "Add this RSS Feed Widget"]
-                      when "chart"
-                        ["piechart", "Add this Chart Widget"]
-                      when "report"
-                        ["report", "Add this Report Widget"]
-                      when "menu"
-                        ["menu", "Add this Menu Widget"]
-        end
-        w.title.gsub!(/'/,"&apos;")     # Need to escape single quote in title to load toolbar
-        opt = root.add_element("option", {"value"=>w.id,"img_src"=>"/images/icons/24/button_#{image}.png"})
-        opt.text = CGI.escapeHTML(w.title)
-      end
-    end
-
     if @available_widgets.blank?
-      opt = root.add_element("option", {"value"=>"","img_src"=>"/images/icons/24/add_widget.png"})
-      opt.text = "No Widgets available to add"
+      @widgets_options = [["No Widgets available to add", "", {"data-icon" => "product product-arrow-right"}]]
+    else
+      @widgets_options = [["Add a Widget", "", {"data-icon" => "product product-arrow-right"}]]
+
+      @available_widgets.each do |w|
+        next if col_widgets.include?(w.id) || !w.enabled
+        image = case w.content_type
+                when "rss"
+                  "fa fa-rss"
+                when "chart"
+                  "product product-chart"
+                when "report"
+                  "product product-report"
+                when "menu"
+                  "fa fa-share-square-o"
+                end
+        w.title.gsub!(/'/, "&apos;")     # Need to escape single quote in title to load toolbar
```

Comment:
Should this be more generic than just an apostrophe check?  Something like `w.title = CGI.escapeHTML(w.title)`